jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java

```diff
@@ -102,15 +162,27 @@ default HttpFields get()
     }
 
     /**
-     * Efficiently take the fields as an Immutable that cannot be changed by any further mutations
-     * to this instance.
-     * @return An immutable version of the fields.
+     * <p>Returns an immutable copy of this {@link HttpFields} instance.</p>
+     *
+     * @return a new immutable copy of this {@link HttpFields} instance
      */
-    default HttpFields takeAsImmutable()
+    default HttpFields asImmutable()
```

Comment:
I'm not sure a default implementation is desirable, as it feels like this should always be a specialized or a delegating implementation.

Reply:
`HttpFields` is a functional interface (only needs to implement `listIterator()`), so it's handy.

Reply:
Ok, I'm fine with that.